### PR TITLE
added browsersupport

### PR DESCRIPTION
**Pull request by svenjens**

### CODE REVIEW

**Comment by robinpiets** [comment]
Bug 1:
image
- Go from Address Page to Question Page.
- Use browser button prev
- Use browser button next
- Get error

**Comment by robinpiets** [comment]
Bug 2:
- Go from Address Page to Question Page.
- Go from Question 1 to Question 2
- Use browser refresh and I am back at Question 1
- Use browser refresh again and I am back at Question 2

**Comment by robinpiets** [comment]
Bug 3:
- Go to Results Page
- Use browser back to go to last question
- Use browser next to go to Results Page
- Use browser back to go to last question and I am on another question

File: apps/client/src/sttr_client/models/checker.js [hunk 1]
```diff
@@ -50,6 +50,28 @@ class Checker {
     return this._stack;
   }
 
+  /**
+   * @returns {Question} -
+   */
```
Comment (afjlambert) [inline apps/client/src/sttr_client/models/checker.js:55]: Can you update the comments?

File: apps/client/src/sttr_client/models/checker.js [hunk 2]
```diff
@@ -50,6 +50,28 @@ class Checker {
...
+  getData() {
+    return this.stack.reduce((acc, question) => {
+      acc[question.id] = question.answer;
+      return acc;
+    }, {});
+  }
+
+  setData(answers) {
```
Comment (afjlambert) [inline apps/client/src/sttr_client/models/checker.js:67]: Can you add comments?

File: apps/client/src/pages/QuestionsPage.js [hunk 3]
```diff
@@ -7,34 +7,50 @@ import withChecker from "../hoc/withChecker";
...
+          //handle back button
+          const prev = checker.previous();
+          setQuestion(prev);
+        }
+      }
+    });
+  }, [locationKeys, checker, history]);
 
   // Update URL based on question text
-  if (!questionSlug || questionSlug !== currSlug) {
```
Comment (afjlambert) [inline apps/client/src/pages/QuestionsPage.js:23]: Why do you remove this check?

File: apps/client/src/pages/ResultsPage.js [hunk 4]
```diff
@@ -75,55 +78,60 @@ const ResultsPage = ({ topic, checker }) => {
...
-                    <Icon
-                      color="secondary"
-                      size={30}
-                      style={{
-                        flexShrink: 0, // IE11 Fix
-                      }}
+            if (
+              typeof question.answer === "boolean" ||
+              typeof question.answer === "string"
+            ) {
```
Comment (afjlambert) [inline apps/client/src/pages/ResultsPage.js]: Why add this check?

**Review by afjlambert — CHANGES_REQUESTED** [review]

File: apps/client/src/context.js [hunk 5]
```diff
@@ -1,3 +1,33 @@
...
+  ? session
+  : { topic: null, address: null, questionId: 0 };
+
+const Context = createContext();
+
+function CheckerProvider(props) {
+  const [data, setData] = useReducer(reducer, defaultValues);
+
+  useEffect(() => {
+    sessionStorage.setItem("mySession", JSON.stringify(data));
```
Comment (robinpiets) [inline apps/client/src/context.js]: It looks like `mySession` came out of a tutorial, can you rename to `vergunningcheck` or something?

File: apps/client/src/index.js [hunk 6]
```diff
@@ -10,7 +10,7 @@ import { ApolloProvider } from "@apollo/react-hooks";
 import { MatomoProvider, createInstance } from "@datapunt/matomo-tracker-react";
 import { GlobalStyle, ThemeProvider, themeColor } from "@datapunt/asc-ui";
 import { createGlobalStyle } from "@datapunt/asc-core";
-import Context, { defaultValues } from "./context";
+import { CheckerProvider } from "./context";
```
Comment (robinpiets) [inline apps/client/src/index.js]: Why do you use `Checker` and not `Check`? We renamed the vergunning~s~**checker** to vergunning**check**. What do you think @afjlambert?

File: apps/client/src/pages/QuestionsPage.js [hunk 7]
```diff
@@ -7,34 +7,50 @@ import withChecker from "../hoc/withChecker";
...
+
+      if (history.action === "POP") {
+        if (locationKeys[1] === location.key) {
+          setLocationKeys(([_, ...keys]) => keys);
+          // Handle forward event
+          const next = checker.next();
+          setQuestion(next);
+        } else {
+          setLocationKeys((keys) => [location.key, ...keys]);
+          //handle back button
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.js]: Nitpicking: Formatting of comments

File: apps/client/src/pages/QuestionsPage.js [hunk 8]
```diff
@@ -45,39 +61,52 @@ const QuestionsPage = ({ topic, checker }) => {
...
     }
-  };
 
-  const onQuestionPrev = () => {
-    if (checker?.stack?.length > 1) {
+    // Go back to Location page
+    if (back && checker?.stack?.length === 1) {
+      return history.push(geturl(routes.address, { slug }));
+    }
+    // Handle back option
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.js]: What means "Handle back option"?

File: apps/client/src/pages/QuestionsPage.test.js [hunk 9]
```diff
@@ -45,10 +45,10 @@ describe("<QuestionsPage />", () => {
   // Mock address
   const { exactMatch } = address[0].result.data.findAddress;
 
-  const Wrapper = () => {
+  const Wrapper = ({questionId}) => {
     const history = createMemoryHistory();
     return (
-      <Context topicMock={topic} addressMock={exactMatch} checker={checker}>
+      <Context topicMock={topic} addressMock={exactMatch} checker={checker} questionId={questionId}>
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.test.js]: Why do you need `questionId` if you only pass `{0}` in the three tests below?

**Review by robinpiets — CHANGES_REQUESTED** [review]
Looking good! Some minor change requests

File: apps/client/src/index.js (hunk 6 above)
Comment (svenjens) [reply apps/client/src/index.js:13]: i Agree!

File: apps/client/src/context.js (hunk 5 above)
Comment (svenjens) [reply apps/client/src/context.js]: I named it mySession with the purpose of maybe sharing later. 
Maybe rename it to vergunningcheckData? 

Also its saved to the sessionStorage, so it is a session.

Comment (robinpiets) [reply apps/client/src/context.js]: `permitData` `vergunningData` `vergunningcheckData` `checkData`, you choose..

BTW, do we want every *activity* to have its own storage key? In that case, we can also add those key-names to the `config`, like: 
```
{
  slug: "dakkapel-plaatsen",
  sessionStorage: "dakkapel-storage",
  ...etc,
}
```

File: apps/client/src/pages/ResultsPage.js [hunk 10]
```diff
@@ -22,14 +22,22 @@ import {
   Change,
 } from "./ResultsPageStyles";
 import Helmet from "react-helmet";
+import Context from "../context";
 
 const ResultsPage = ({ topic, checker }) => {
   const history = useHistory();
+  const context = useContext(Context);
   const { slug } = topic;
+  debugger;
```
Comment (robinpiets) [inline apps/client/src/pages/ResultsPage.js]: debugger; ...

File: apps/client/src/context.js [hunk 11]
```diff
@@ -1,3 +1,33 @@
...
+  if (newData === null) {
+    sessionStorage.removeItem("vergunningCheckData");
+    return defaultValues;
+  }
+  return { ...data, ...newData };
+};
+
+const defaultValues = session
+  ? session
+  : { topic: null, address: null, questionId: 0 };
```
Comment (robinpiets) [inline apps/client/src/context.js]: Why is this called `questionId` and not `questionIndex`? The real `questionId` is a hash and not a number

File: apps/client/src/pages/QuestionsPage.js [hunk 12]
```diff
@@ -45,39 +62,58 @@ const QuestionsPage = ({ topic, checker }) => {
...
-      const next = checker.next();
+    }
 
+    if (!needContactPermits() && !back) {
+      const next = context.resultsShown
+        ? checker.stack[context.questionId]
+        : checker.next();
+
+      console.log("results shown", context.resultsShown);
+      console.log("next", next);
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.js]: Remove logs

File: apps/client/src/pages/QuestionsPage.js [hunk 13]
```diff
@@ -45,39 +62,58 @@ const QuestionsPage = ({ topic, checker }) => {
...
+
+        context.setData({ data: checker.getData() });
         history.push(geturl(routes.results, { slug }));
-      } else {
+      }
+      if (next) {
         // Go to Next question
         setQuestion(next);
+        context.setData({ questionId: context.questionId + 1 });
+        console.log(next);
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.js]: Remove logs

File: apps/client/src/pages/QuestionsPage.test.js [hunk 14]
```diff
@@ -81,7 +80,7 @@ describe("<QuestionsPage />", () => {
 
   it("navigates correctly between Question Page, Conclusion Page and Address Page", async () => {
     const { container, getByTestId, findByText, findByTestId } = render(
-      <Wrapper />
+      <Wrapper questionId={0} />
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.test.js]: Why is this `{0}` and the test below `{1}`?

File: apps/client/src/pages/QuestionsPage.test.js [hunk 15]
```diff
@@ -119,7 +118,7 @@ describe("<QuestionsPage />", () => {
 
   it("navigates correctly between different questions and Results Page", async () => {
     const { container, getByTestId, findByText, findByTestId } = render(
-      <Wrapper />
+      <Wrapper questionId={1} />
```
Comment (robinpiets) [inline apps/client/src/pages/QuestionsPage.test.js]: Why is this `{1}` and the test above `{0}`?

File: apps/client/src/pages/ResultsPage.js (hunk 4 above)
Comment (svenjens) [reply apps/client/src/pages/ResultsPage.js]: To make sure the questions are always set.

**Pull request closed without merging**